models/DataObject/ClassDefinition/Data/UrlSlug.php

```diff
@@ -286,6 +290,42 @@ public function save($object, $params = [])
                 }
             }
         }
+
+        // check for previous slugs and create redirects
+        if(is_array($data)) {
+            foreach($data as $slug) {
+                if($previousSlug = $slug->getPreviousSlug()) {
```

Comment:
```
if($previousSlug !== $slug->getPreviousSlug()) {
    continue;
}
```

Reply:
I guess you mean `if($previousSlug === $slug->getSlug()) {` right?